### PR TITLE
fix(sync): only parse Markdown front matter at document start

### PR DESCRIPTION
**Pull request by FeishuZheng**
> [!IMPORTANT]
> **AI-assisted contribution disclosure:** This fix was implemented by OpenAI Codex under the contributor's direction. The contributor reproduced the original truncation behavior and reviewed the resulting patch.
>
> **Scope and risk:** This revision intentionally keeps the existing regular expression, YAML parser, and newline behavior. It only rejects a match when document content precedes it. This avoids introducing new parsing semantics, while retaining existing limitations such as leading-BOM/blank-line handling and malformed-front-matter behavior.

## Problem

`getMDStatusFromContent` searches globally for a block delimited by `---`. In a headerless Markdown document, horizontal rules in the body can therefore become the first match. The old code then slices from the beginning by the matched block's length, dropping an unrelated prefix.

## Change

Keep the existing parser and accept its first match only when the document starts with that match:

```ts
if (result && contentRaw.startsWith(result[0])) {
```

The production-code diff is one condition. Two focused tests cover the reported regression and confirm that valid front matter at the document start is still parsed.

## Validation

- `npx tsc --noEmit`
- `npx prettier --check src/modules/sync/api.ts test/tests/sync-mdStatus.spec.ts`
- `npx eslint src/modules/sync/api.ts test/tests/sync-mdStatus.spec.ts --ext .ts`
- `git diff --check`
- `npm run build` (production XPI, version `3.3.0-beta.6`)
- exact-regex check against a 39,570-character related report containing 13 body matches: the first match begins at offset 679, and the new guard preserves all 39,570 characters

`npm test` on the latest beta.6 base currently stops at the unrelated existing test `follows the default-mode and toggle-button preferences`: beta.6 changed `editor.showMarkdownToggle` to `false`, while that test still expects the toggle to be visible by default. With only that upstream test temporarily skipped locally, the remaining suite completed with **50 passed**, including both tests added here. The skip was restored and is not part of this PR.

Related to #1600. The symptom overlaps, but this PR does not assume that every reported approximately-32k truncation case has the same root cause.

### CODE REVIEW

File: src/modules/sync/api.ts [hunk 1]
```diff
@@ -109,23 +109,78 @@ function getSyncStatus(noteId?: number): SyncStatus {
...
   };
-  if (result) {
-    const yaml = result[0].replace(/---/g, "");
-    ret.content = contentRaw.slice(result[0].length);
-    try {
-      ret.meta = YAML.parse(yaml);
-    } catch (e) {
-      ztoolkit.log(e);
+
+  const openingFence = contentRaw.match(/^---[ \t]*\n/);
```
Comment (windingwind) [inline src/modules/sync/api.ts]: This is over complicated. I would prefer simply checking if there's any content before the first matched yaml, instead of lines of loop and check...

Comment (FeishuZheng) [reply src/modules/sync/api.ts]: Thanks — simplified as suggested. The existing regex and YAML parser are now unchanged; the only production change is `result && contentRaw.startsWith(result[0])`. I also reduced the tests from 27 edge cases to two focused cases: the regression and valid front matter at the document start. The branch is rebased onto the current `master`.

**Review by windingwind — APPROVED** [review]
